### PR TITLE
Refactor: Remove unnecessary dependency on dubbogo/grpc-go 

### LINKED ISSUE #2560: Avoid unnecessary dependency on `dubbogo/grpc-go`
Details in https://github.com/apache/dubbo-go/pull/2545#discussion_r1439132523

How about we make a copy of the definitions needed in dubbogo/grpc-go? I think it would be better to not depend on dubbogo/grpc-go except for some compatibility needs.

### PR DESCRIPTION
**Pull request by shikharish**
Closes #2560

### CODE REVIEW

File: protocol/triple/triple_protocol/codec.go [hunk 1]
```diff
@@ -337,3 +338,193 @@ func errNotProto(message interface{}) error {
 	}
 	return fmt.Errorf("%T doesn't implement proto.Message", message)
 }
+
+// Definitions from dubbogo/grpc-go
+func GetArgType(v interface{}) string {
```
Comment (DMwangnima) [inline protocol/triple/triple_protocol/codec.go:343]: For now, this function does not need to be exposed. We would expose it if needed in the future.

File: protocol/triple/triple_protocol/codec.go [hunk 2]
```diff
@@ -337,3 +338,193 @@ func errNotProto(message interface{}) error {
...
+			return "java.util.List"
+		case reflect.Map: // Enter here, map may be map[string]int
+			return "java.util.Map"
+		default:
+			return ""
+		}
+	}
+}
+
+func ReflectResponse(in interface{}, out interface{}) error {
```
Comment (DMwangnima) [inline protocol/triple/triple_protocol/codec.go:434]: ditto.

File: protocol/triple/triple_protocol/codec.go [hunk 3]
```diff
@@ -337,3 +338,193 @@ func errNotProto(message interface{}) error {
...
+		return CopyMap(inValue, outValue)
+	default:
+		hessian.SetValue(outValue, inValue)
+	}
+
+	return nil
+}
+
+// CopySlice copy from inSlice to outSlice
+func CopySlice(inSlice, outSlice reflect.Value) error {
```
Comment (DMwangnima) [inline protocol/triple/triple_protocol/codec.go:468]: ditto.

File: protocol/triple/triple_protocol/codec.go [hunk 4]
```diff
@@ -337,3 +338,193 @@ func errNotProto(message interface{}) error {
...
+				inSliceValue.Type().String(), outSlice.Type().String())
+		}
+		outSlice.Index(i).Set(inSliceValue)
+	}
+
+	return nil
+}
+
+// CopyMap copy from in map to out map
+func CopyMap(inMapValue, outMapValue reflect.Value) error {
```
Comment (DMwangnima) [inline protocol/triple/triple_protocol/codec.go:496]: ditto.

File: protocol/triple/triple_protocol/internal/interoperability/interop.triple_wrapper.proto [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interoperability;
+
+option go_package = "../interoperability";
```
Comment (DMwangnima) [inline protocol/triple/triple_protocol/internal/interoperability/interop.triple_wrapper.proto:20]: Specify go_package with "dubbo.apache.org/dubbo-go/v3/protocol/triple/triple_protocol/internal/interoperability;interoperability" and use ```protoc --go_out=. --go_opt=paths=source_relative``` is better?

**Review by DMwangnima — CHANGES_REQUESTED** [review]
BTW, pls fix the CI.

**Review by DMwangnima — APPROVED** [review]
LGTM.

**Comment by DMwangnima** [comment]
@shikharish You really did a good job! Are you willing to participate in our samples transformation work? We'd love it if you could get involved.

**Comment by shikharish** [comment]
Yes, I would love to!

**Comment by DMwangnima** [comment]
@shikharish Great! Can you give me your email address since we could contact you fast. This is my email: 
dmdwangyuxuan@gmail.com. And this task would be published in https://github.com/apache/dubbo-go-samples. I would let you know if this issue published successfully.

**Comment by shikharish** [comment]
@DMwangnima Here's my email: shikharish05@gmail.com
Thank you!

**Comment by chickenlj** [comment]
@shikharish Welcome!

As you might have already noticed, the API in the main branch differs from that in the previous versions. According to the plan we have with the 3.2.0 release, we will provide a lightweight implemented, and more front-end-friendly RPC protocol that is 100% compatible with gRPC. We will also have the programming API updated with the new protocol, which should be easier and straightforward for users to use and understand.

With most of the tasks done in the main branch of apache/dubbo, it becomes the priority task to update the examples in apache/dubbo-go-samples to adapt to the new API, so it can work well as demo showcases for users and integration tests for ourselves.

**Comment by chickenlj** [comment]
@DMwangnima has already finished the `helloworld` and `registry` parts in apache/dubbo-go-samples, which can be used as a reference usage of the new API.

**Comment by shikharish** [comment]
@chickenlj So should I just pick one part and update it to use new API? Or as @DMwangnima said he would let me know about it?

**Comment by chickenlj** [comment]
@shikharish Please see the comment I left above, I think you can start with some of the above features now.

I opened a PR to relocate the old examples into the compatibility directory https://github.com/apache/dubbo-go-samples/pull/666. Those examples will only work as IT cases and references for creating new ones.

**Comment by shikharish** [comment]
@chickenlj Thank you. I will update soon!

**Pull request merged**